qt/widgets/mplcpp/src/FigureEventFilter.cpp

```diff
@@ -0,0 +1,24 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidQtWidgets/MplCpp/FigureEventFilter.h"
+
+#include <QResizeEvent>
+
+namespace MantidQt::Widgets::MplCpp {
+
+bool FigureEventFilter::eventFilter(QObject *obj, QEvent *ev) {
+  if (ev->type() == QEvent::Resize) {
+    auto const size = static_cast<QResizeEvent *>(ev)->size();
+    if (size.width() < 1 || size.height() < 1) {
```

Comment:
FYI, could instead use the `QSize::isEmpty()` function:
https://doc.qt.io/qt-5/qsize.html#isEmpty